### DISCUSSION TITLE
Rad Pro: Sign the firmware

### DISCUSSION THREAD

**Discussion by alfmck** [Gissio/radpro#49] [Q&A] [answered]
I tried to make radpro-fnirsi-gc01-ch32f103rb-2.0rc2.bin (original size is 48 KB) from source code (zip).
after compiling  in .pi0.../fnisrsi-gc01-ch32f103r8 i got firmware. Bin file which is 42 KB size.
When I flash this new firmware in my GC-01, i get starting message "Rad Pro 2.0rcs2 check sum error". The GC-01 registers high rates radiation and beeps of course.
Could somebody explain where I made a mistake. Also I didn't understand the meaning  of "Once you've built the firmware, sign the resulting binaries with the tools/radpro-sign.py script.:. I try to compile this script, but it didn't" change the firmware.bin file  and only creates sconsign311.dblite only.

**Comment by Gissio** [comment, marked as answer]
You did not make any mistake. The firmware is signed so that, in case of high radiation, you get notified that the firmware's integrity has been compromised.

To sign the firmware, make sure you install Python and the dependencies ([https://github.com/Gissio/radpro/blob/main/docs/users.md#radpro-tool](https://github.com/Gissio/radpro/blob/main/docs/users.md#radpro-tool)). Next, start a terminal, go to the `tools` folder, and from there start the `radpro-sign.py` script:

    python radpro-sign.py

 A couple of `.bin` files should appear in the `tools` folder.

**Reply by alfmck** [reply]
Everything OK. Thank You, Gissio very much for help.
